### PR TITLE
Edit comment date to local users date

### LINKED ISSUE #7561: Error in date display on comment
**Describe the bug**

Date tag on a comment is "Apr 27", but on hover the text is "Tuesday, 28 April 2020, 00:01:19".

In HTML, datetime content is `2020-04-27T22:01:19Z` :thinking: 

**To Reproduce**

Hover date of [this](https://dev.to/lofiandcode/comment/o8k9) comment.

**Expected behavior**

Date on tag & on hover should be the same.

**Screenshots**

image

image

**Desktop (please complete the following information):**

- OS: Xubuntu
- Browser: Firefox
- Version: 75.0

### PR DESCRIPTION
**Pull request by eclectic-coding**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
I tested this change by changing me system date from UTC -4 to UTC +2 and created a post and comment in the local environment. Then I changed my system time back. After this fix, the dates display properly. This change implements the new `local_date` helper. 

## Related Tickets & Documents
Fixes #7561 

## QA Instructions, Screenshots, Recordings
Screenshot from 2020-07-29 22-00-37

Hovering over the comment date, should display a formatted date that builds on the displayed short date for the comment.

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by eclectic-coding** [comment]
Thanks I will work on this later today.

**Comment by msarit** [comment]
Hey @eclectic-coding! :wave:
I wanted to check-in and see if you're still working on this PR. Please let us know if you need help, or have any questions. Thanks!

**Comment by eclectic-coding** [comment]
Yes I will get back into today. I think all that is left is the tests.

**Comment by citizen428** [comment]
Hi @eclectic-coding, not meaning to nag, just checking in if there's anything we can do to help you push this over the finish line? Thanks for your efforts so far!

**Comment by eclectic-coding** [comment]
@citizen428  Sorry for the delay. Just got forem setup on my new hardware and will get into again today. If I need help I will connect back.

**Comment by eclectic-coding** [comment]
@citizen428 Actually I do need some help. I am not real strong on test and have started this one, but can't figure out how to proceed. If you could give me some input I would appreciate it.

File: .env [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Development options
```
Comment (citizen428) [inline .env:1]: You accidentally commited this file, it shouldn't be here.

File: db/schema.rb [hunk 2]
```diff
@@ -1197,6 +1197,7 @@
     t.boolean "export_requested", default: false
```
Comment (citizen428) [inline db/schema.rb:1197]: This PR has migration, so there shouldn't be any schema changes.

**Comment by citizen428** [comment]
@eclectic-coding No worries, You could use [Timecop](https://github.com/travisjeffery/timecop)'s `Timecop.freeze` method to run the assertion at two different times and check that the returned timestamps are what you'd expect.

File: db/schema.rb [hunk 3]
```diff
@@ -1,1358 +0,0 @@
-# This file is auto-generated from the current state of the database. Instead
```
Comment (citizen428) [inline db/schema.rb:1]: @eclectic-coding This shouldn't be part of this PR.

**Review by maestromac — CHANGES_REQUESTED** [review]
Apologies for spotting this just now. Unfortunately, this fix won't work because comments are cached on fastly. This means those dates are UTC, and we provide a raw DateTime for mouse hover. That DateTime is automatically converted by the browser to the user's current timezone.

A solution will have to be javascript on the frontend to convert those displayed times.

**Comment by msarit** [comment]
@eclectic-coding Hey there! 👋🏾  Thank you for your work on this PR!

There are a few conflicting files to resolve, and Mac suggested [implementing your changes via javascript on the frontend](https://github.com/forem/forem/pull/9569#pullrequestreview-605608466) (since comments are cached).

Let us know if you need any help completing this implementation.

**Comment by citizen428** [comment]
Since no work has happened on this PR in almost 2 months I'll close it now. @eclectic-coding Please feel free to reopen it if you pick it up again. Thanks for your contribution!

**Pull request closed without merging**